autopilot/hostinfo.go

```diff
@@ -66,16 +67,46 @@ func (c *contractor) HostInfo(ctx context.Context, hostKey types.PublicKey) (api
 	}, nil
 }
 
+func (c *contractor) hostInfoFromCache(ctx context.Context, host hostdb.Host) (hi hostInfo, found bool) {
+	// grab host details from cache
+	c.mu.Lock()
+	hi, found = c.cachedHostInfo[host.PublicKey]
+	storedData := c.cachedDataStored[host.PublicKey]
+	minScore := c.cachedMinScore
+	c.mu.Unlock()
+
+	// return early if the host info is not cached
+	if !found {
+		return
+	}
+
+	// try and refresh the host info if it got scanned in the mean time, this
+	// inconsistency would resolve itself but tyring to update it here improves
+	// first time user experience
```

Comment:
I think this is a bit of a roundabout way of doing it. Why not have the scanner directly update the cache upon a successful scan?

Reply:
I didn't want to tie in the `contractor` with the `scanner`... I considered doing that but it makes them coupled in a way that I figured best to avoid? I also considered pulling out the cache but didn't really seem worth it.

Reply:
The thing that makes this not feel very clean to me is that we are keeping a cache that checks for an edge case and then performs an API call to update some data without actually refreshing itself. If we go that route I'd rather do something like
```go
if host.Scanned && hi.UnusableResult.notcompletingscan > 0 {
 // handle edge case where host gets scanned between maintenance loop iterations
  host.UnusableResult.notcompletingscan = 0 
}
hi = hostInfo{
  Usable: hi.UnusableResult.IsUsable(),
  UnusableResult: hi.UnusableResult,
}
```
without the call to `isUsableHost`. That way we still only rely on cached values.

Reply:
Would you agree to updating the cache instead? Manipulating `notcompletingscan` would mark the host as usable, only to be marked as unusable the next time it gets scanned. So as a user I would see a bunch of green checkmarks that flip red on the next scan.

Another option is to have both `scanner` and `contractor` use a shared cache. Also fine imo? 
Although I would prefer simply updating the cache here, not sure why I didn't do that in the first place.

Reply:
I'd usually go for a shared cache but in the spirit of potentially replacing this with moving the values to the bus it's probably less intrusive to update the cache here for now. Once it's in the bus, we will essentially get the shared cache implicitly.